apps/teams-test-app/src/components/privateApis/PrivateAPIs.tsx

```diff
@@ -1,86 +1,90 @@
 import { registerUserSettingsChangeHandler, uploadCustomApp, UserSettingTypes } from '@microsoft/teams-js';
 import React, { ReactElement } from 'react';
 
-import { generateJsonParseErrorMsg, noHostSdkMsg } from '../../App';
-import BoxAndButton from '../BoxAndButton';
+import { noHostSdkMsg } from '../../App';
+import { ApiWithoutInput, ApiWithTextInput } from '../utils';
 
-const PrivateAPIs = (): ReactElement => {
-  const [uploadCustomAppRes, setUploadCustomAppRes] = React.useState('');
-  const [registerUserSettingsChangeHandlerRes, setRegisterUserSettingsChangeHandlerRes] = React.useState('');
+const getUserSettingTypesFromInput = (input: string[]): UserSettingTypes[] => {
+  const ret: UserSettingTypes[] = [];
+  if (input.includes('fileOpenPreference')) {
+    ret.push(UserSettingTypes.fileOpenPreference);
+  }
+  if (input.includes('theme')) {
+    ret.push(UserSettingTypes.theme);
+  }
 
-  const getUserSettingTypesFromInput = (input: string[]): UserSettingTypes[] => {
-    const ret: UserSettingTypes[] = [];
-    if (input.includes('fileOpenPreference')) {
-      ret.push(UserSettingTypes.fileOpenPreference);
-    }
-    if (input.includes('theme')) {
-      ret.push(UserSettingTypes.theme);
-    }
+  return ret;
+};
 
-    return ret;
-  };
+const RegisterUserSettingsChangeHandler = (): React.ReactElement =>
+  ApiWithTextInput<string[]>({
+    name: 'registerUserSettingsChangeHandler',
+    title: 'Register User Settings Change Handler',
+    onClick: {
+      validateInput: input => {
+        if (!input || !Array.isArray(input) || input.length === 0 || input.find(x => typeof x !== 'string')) {
+          throw new Error('input has to be an array of strings with at least one element.');
+        }
+      },
+      submit: async (input, setResult) => {
+        // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+        const handler = (settingType: UserSettingTypes, value: any): void => {
+          setResult(`Success. settingType: ${settingType}, value: ${value}`);
+        };
 
-  const acceptFile = (files: FileList | null): void => {
-    if (files) {
-      if (files.length != 1) {
-        throw new Error('There should be exactly one file uploaded.');
-      }
-      const onComplete = (status: boolean, reason?: string): void => {
-        let message = `status: ${status}`;
-        if (reason) {
-          message = message + `, reason: ${reason}`;
+        registerUserSettingsChangeHandler(getUserSettingTypesFromInput(input), handler);
+        return 'registerUserSettingsChangeHandler()' + noHostSdkMsg;
```

Comment:
Similar to the other comment, I think we might want "called" here or use generateRegistrationMsg()